packages/vite/src/node/optimizer/scan.ts

```diff
@@ -226,7 +220,7 @@ function esbuildScanPlugin(
 
       build.onLoad(
         { filter: virtualModuleRE, namespace: 'html' },
```

Comment:
But if we do that, this filter never catches it, so we didn't load the virtual module. Instead, the `onLoad()` below this block of code captures it instead.